File: modules/076-skills-management-system/tools/scripts/cmd/initrepo.go
package cmd

import (
	"flag"
	"fmt"
	"os"
	"path/filepath"
)

// RunInitRepo handles the `skills init-repo <folder-name>` command.
func RunInitRepo(args []string) {
	fs := flag.NewFlagSet("init-repo", flag.ExitOnError)
	fs.Usage = func() {
		fmt.Print(`Initialize a new skills repository with base structure.

Creates a Git-initialized folder with:
  .manifest/_global.json      — global skills config
  .manifest/group-1.json      — example group config
  .manifest/sub-group.json    — example sub-config
  skills-cli/                 — skill: CLI usage, creating skills, IDE integration

Usage:
  skills init-repo <folder-name>

Examples:
  skills init-repo my-skills-repo
  skills init-repo ../shared-skills
`)
	}

	if err := fs.Parse(args); err != nil {
		os.Exit(1)
	}

	positional := fs.Args()
	if len(positional) == 0 {
		fmt.Fprintln(os.Stderr, "Error: folder name is required")
		fs.Usage()
		os.Exit(1)
	}

	folderName := positional[0]

	if _, err := os.Stat(folderName); err == nil {
		fmt.Fprintf(os.Stderr, "Error: folder %q already exists\n", folderName)
		os.Exit(1)
	}

	fmt.Printf("→ Creating skills repository at %s ...\n", folderName)

	// Create directory structure
	dirs := []string{
		filepath.Join(folderName, ".manifest"),
		filepath.Join(folderName, "skills-cli"),
	}
	for _, d := range dirs {
		if err := os.MkdirAll(d, 0755); err != nil {
			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
			os.Exit(1)
		}
	}

	// Write manifest files
	writeFile(filepath.Join(folderName, ".manifest", "_global.json"), globalJSON)
	writeFile(filepath.Join(folderName, ".manifest", "group-1.json"), group1JSON)
	writeFile(filepath.Join(folderName, ".manifest", "sub-group.json"), subGroupJSON)

	// Write skills-cli skill (read SKILL-CLI.md from disk as source of truth)
	skillCliContent := loadSkillCliMd()
	writeFile(filepath.Join(folderName, "skills-cli", "SKILL.md"), skillCliContent)
	writeFile(filepath.Join(folderName, "skills-cli", "info.json"), skillsCLIInfo)

	// Write .gitignore
	writeFile(filepath.Join(folderName, ".gitignore"), repoGitignore)

	fmt.Println("  ✓ Files created")
	fmt.Printf("\n✅ Skills repository initialized at %s\n", folderName)
	fmt.Println("\nNext steps:")
	fmt.Printf("  cd %s\n", folderName)
	fmt.Println("  git init && git add . && git commit -m \"init: skills repository\"")
	fmt.Println("  # Then push to your Git hosting")
}

func writeFile(path, content string) {
	if err := os.WriteFile(path, []byte(content), 0644); err != nil {
		fmt.Fprintf(os.Stderr, "Error writing %s: %v\n", path, err)
		os.Exit(1)
	}
}

const globalJSON = `{
  "skills": [
    "skills-cli"
  ]
}
`

const group1JSON = `{
  "skills": [],
  "sub-configs": ["sub-group"]
}
`

const subGroupJSON = `{
  "skills": [],
  "sub-configs": []
}
`

const repoGitignore = `# Skills repo .gitignore
`

// loadSkillCliMd reads SKILL-CLI.md relative to the executable.
func loadSkillCliMd() string {
	exePath, err := os.Executable()
	if err == nil {
		candidate := filepath.Join(filepath.Dir(exePath), "..", "SKILL-CLI.md")
		if data, readErr := os.ReadFile(candidate); readErr == nil {
			return string(data)
		}
	}
	return "# Skills CLI\n\nSee SKILL-CLI.md in the skills-cli package for full reference.\n"
}

const skillsCLIInfo = `{
  "description": "Skills CLI reference: commands, creating skills, IDE integration (VSCode/Copilot, Cursor, Claude Code).",
  "owner": "your-name@example.com"
}
`

File: modules/076-skills-management-system/tools2/src/commands/initrepo.ts
import * as fs from 'fs';
import * as path from 'path';

export function runInitRepo(args: string[]): void {
  if (args.includes('--help') || args.includes('-h')) {
    printInitRepoHelp();
    return;
  }

  const positional = args.filter(a => !a.startsWith('--'));
  if (positional.length === 0) {
    console.error('Error: folder name is required');
    printInitRepoHelp();
    process.exit(1);
  }

  const folderName = positional[0];

  if (fs.existsSync(folderName)) {
    console.error(`Error: folder "${folderName}" already exists`);
    process.exit(1);
  }

  console.log(`→ Creating skills repository at ${folderName} ...`);

  // Create directory structure
  const dirs = [
    path.join(folderName, '.manifest'),
    path.join(folderName, 'skills-cli'),
  ];
  for (const d of dirs) {
    fs.mkdirSync(d, { recursive: true });
  }

  // Write manifest files
  writeFile(path.join(folderName, '.manifest', '_global.json'), GLOBAL_JSON);
  writeFile(path.join(folderName, '.manifest', 'group-1.json'), GROUP1_JSON);
  writeFile(path.join(folderName, '.manifest', 'sub-group.json'), SUBGROUP_JSON);

  // Write skills-cli skill (read SKILL-CLI.md from package root as source of truth)
  const skillCliContent = loadSkillCliMd();
  writeFile(path.join(folderName, 'skills-cli', 'SKILL.md'), skillCliContent);
  writeFile(path.join(folderName, 'skills-cli', 'info.json'), SKILLS_CLI_INFO);

  // Write .gitignore
  writeFile(path.join(folderName, '.gitignore'), '# Skills repo .gitignore\n');

  console.log('  ✓ Files created');
  console.log(`\n✅ Skills repository initialized at ${folderName}`);
  console.log('\nNext steps:');
  console.log(`  cd ${folderName}`);
  console.log('  git init && git add . && git commit -m "init: skills repository"');
  console.log('  # Then push to your Git hosting');
}

function writeFile(filePath: string, content: string): void {
  fs.writeFileSync(filePath, content, 'utf8');
}

function printInitRepoHelp(): void {
  console.log(`Initialize a new skills repository with base structure.

Creates a folder with:
  .manifest/_global.json      — global skills config
  .manifest/group-1.json      — example group config
  .manifest/sub-group.json    — example sub-config
  skills-cli/                 — skill: CLI usage, creating skills, IDE integration

Usage:
  skills init-repo <folder-name>

Examples:
  skills init-repo my-skills-repo
  skills init-repo ../shared-skills
`);
}

const GLOBAL_JSON = `{
  "skills": [
    "skills-cli"
  ]
}
`;

const GROUP1_JSON = `{
  "skills": [],
  "sub-configs": ["sub-group"]
}
`;

const SUBGROUP_JSON = `{
  "skills": [],
  "sub-configs": []
}
`;

function loadSkillCliMd(): string {
  const candidates = [
    path.join(__dirname, '..', '..', 'SKILL-CLI.md'),  // from dist/commands/
    path.join(__dirname, '..', 'SKILL-CLI.md'),
  ];
  for (const p of candidates) {
    if (fs.existsSync(p)) {
      return fs.readFileSync(p, 'utf8');
    }
  }
  return '# Skills CLI\n\nSee SKILL-CLI.md in the skills-cli package for full reference.\n';
}

const SKILLS_CLI_INFO = `{
  "description": "Skills CLI reference: commands, creating skills, IDE integration (VSCode/Copilot, Cursor, Claude Code).",
  "owner": "your-name@example.com"
}
`;
